### PR TITLE
F# Find references deadlock

### PR DESCRIPTION
**Pull request by nosami**
Fixes https://bugzilla.xamarin.com/show_bug.cgi?id=57925

To test -

- `git clone https://github.com/fsharp/FsAutoComplete.git`
- `cd FsAutoComplete`
- `./build.sh`
- Open up the solution in VSfM
- Wait for tooltips to appear
- Run Find references on any symbol

### CODE REVIEW

**Comment by nosami** [comment]
Build lane - https://code.internalx.com/Wrench/BuildDetails?buildId=d91f2ef57862fc590102f4285925a69

File: main/external/fsharpbinding/MonoDevelop.FSharpBinding/Services/LanguageService.fs [hunk 1]
```diff
@@ -276,6 +276,12 @@ type LanguageService(dirtyNotify, _extraProjectInfo) as x =
         //cache 50 project infos, then start evicting the least recently used entries
         ref (ExtCore.Caching.LruCache.create 50u)
 
+    let optionsForDependentProject p =
```
Comment (Therzok) [inline main/external/fsharpbinding/MonoDevelop.FSharpBinding/Services/LanguageService.fs]: This commit is also contained in https://github.com/mono/monodevelop/pull/2736

File: main/external/fsharpbinding/MonoDevelop.FSharpBinding/Services/LanguageService.fs [hunk 2]
```diff
@@ -324,11 +324,24 @@ type LanguageService(dirtyNotify, _extraProjectInfo) as x =
         //                      opts.ProjectFileName opts.ProjectFileNames opts.ProjectOptions opts.IsIncompleteTypeCheckEnvironment opts.UseScriptResolutionRules)
         Some opts
 
-    member x.GetProjectOptionsFromProjectFile(project:DotNetProject, ?referencedAssemblies) =
-        let getReferencedAssemblies() =
-            retry { return (project.GetReferencedAssemblies(CompilerArguments.getConfig())).Result }
+    member x.GetProjectFromFileName projectFile =
+        IdeApp.Workspace.GetAllProjects()
```
Comment (Therzok) [inline main/external/fsharpbinding/MonoDevelop.FSharpBinding/Services/LanguageService.fs]: `Workspace.GetProjectContainingFile (FilePath)`

File: main/external/fsharpbinding/MonoDevelop.FSharpBinding/Services/LanguageService.fs (hunk 1 above)
Comment (Therzok) [inline main/external/fsharpbinding/MonoDevelop.FSharpBinding/Services/LanguageService.fs]: Nevermind, this targets 15.3.

File: main/external/fsharpbinding/MonoDevelop.FSharpBinding/Services/LanguageService.fs (hunk 2 above)
Comment (nosami) [inline main/external/fsharpbinding/MonoDevelop.FSharpBinding/Services/LanguageService.fs]: 👍

**Pull request merged**